packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js

```diff
@@ -37,36 +41,104 @@ async function saveBundle(bundle) {
     console.log(`Saved bundle to FHIR store!`);
     return ret;
   } catch (error) {
-    console.error(`Could not save bundle: ${error.response.body}`);
-
-    throw new HapiError('Could not save bundle to hapi server!');
+    console.error(`Could not save bundle: ${error.message}`);
+    throw new Error('Could not save bundle to HAPI server!');
   }
 }
 
+async function fetchDataFromUrl(url, maxRetries = 3, delay = 2000) {
+  let attempt = 0;
 
-async function processBundles() {
-  const facilityData = require('./facility_bundle.json');
-  const firstLocationData = require('./location_bundle_part1.json');
-  const secondLocationData = require('./location_bundle_part2.json');
+  while (attempt < maxRetries) {
+    try {
+      const response = await fetch(url);
+      
+      if (!response.ok) {
+        console.warn(`Attempt ${attempt + 1}: Request failed with status ${response.status}`);
+      } else {
+        const data = await response.json();
+        return data;
+      }
+    } catch (error) {
+      console.error(`Attempt ${attempt + 1}: An error occurred: ${error.message}`);
+    }
 
-  try {
-    await saveBundle(facilityData); // Call saveBundle function with the parsed bundle
-    console.log(`Seeder successfully finished execution!`);
-  } catch (error) {
-    console.error(`Error processing bundle: ${error.message}`);
-    process.exit(1);
+    attempt++;
+
+    if (attempt < maxRetries) {
+      // Wait for a specified delay before retrying
+      await new Promise(resolve => setTimeout(resolve, delay));
+    }
   }
 
-  try {
-    await saveBundle(firstLocationData); // Call saveBundle function with the parsed bundle
-    console.log(`Seeder successfully finished execution!`);
-  } catch (error) {
-    console.error(`Error processing bundle: ${error.message}`);
-    process.exit(1);
+  return null;
+}
+
+async function processBundleEntriesAndGenerateTransactionBundle(data) {
+  const resources = [];
+
+  for (const item of data.entry) {
+    // Fetch the resource from the URL
+    let resource = await fetchDataFromUrl(item.fullUrl);
+
+    // Assign identifiers if the resource is a Location or Organization
+    if (!resource && item.resource.resourceType === 'Location') {
+      resource = item.resource;
+      resource.identifier = [{
+        use: "official",
+        system: "http://moh.bw.org/ext/identifier/mfl-code",
```

Comment:
Let us make the url a configurable env variable.

Reply:
Those are code systems from the MFL.

Reply:
So they can not change?